### PR TITLE
FWF-4012:[Feature] - Migration-UI and API integration

### PR DESCRIPTION
**Pull request by Josephalexantony-aot**
# Issue Tracking

JIRA: https://aottech.atlassian.net/browse/FWF-4012
Issue Type: BUG/ FEATURE
# Screenshots (if applicable)
image
image

### CODE REVIEW

**Comment by kiran-aot** [comment]
@Josephalexantony-aot  please fix sonar cloud issues

**Review by abilpraju-aot — APPROVED** [review]

File: forms-flow-web/src/apiManager/services/FormServices.js [hunk 1]
```diff
@@ -25,6 +25,12 @@ export const formUpdate = (form_id,formData) => {
   return RequestService.httpPUTRequest(`${API.FORM_DESIGN}/${form_id}`, formData);
 };
 
+export const processMigrate = (migrationData) => {
+  const migrationUrl = replaceUrl(API.PROCESS_MIGRATE);
```
Comment (shuhaib-aot) [inline forms-flow-web/src/apiManager/services/FormServices.js:29]: is there anything to replace?

File: forms-flow-web/src/apiManager/services/processServices.js [hunk 2]
```diff
@@ -295,16 +295,20 @@ export const getFormProcesses = (formId, ...rest) => {
...
-    let url = replaceUrl(api, "<process_key>", processKey);
-
-    if (tenant_key) {
-      url = url + `?tenantId=${tenant_key}`;
-    }
+export const getProcessDetails = (processKey, tenant_key = null, mapperId = null) => {
+  const api = API.GET_PROCESS_XML;
+  let url = replaceUrl(api, "<process_key>", processKey);
+  // Append tenant_key if available
+  if (tenant_key) {
```
Comment (shuhaib-aot) [inline forms-flow-web/src/apiManager/services/processServices.js:302]: you can optimize code like this , this is a sudo code 
```
const params = [];
if(tenant_key) params.push(`tenantId=${tenant_key}`)
if(mapperId) params.push(`mapperId=${mapperId}`)
if(params.length) url+=`?${params.join("&")`
```

File: forms-flow-web/src/components/Form/EditForm/FlowEdit.js [hunk 3]
```diff
@@ -75,16 +83,42 @@ const FlowEdit = forwardRef(({ isPublished = false, CategoryType, setWorkflowIsC
...
+
+  const handleMigraion = () => {
+    setIsMigraionLoading(true);
+    const migrationData = {
+      mapperId: mapperId,
+      processKey: processData.processKey
+    };
+    processMigrate(migrationData)
+      .then(() => {
+        dispatch(push(`${redirectUrl}formflow`));
```
Comment (shuhaib-aot) [inline forms-flow-web/src/components/Form/EditForm/FlowEdit.js:111]: Is this the correct behavior? If someone attempts to migrate within a form, completes the task, and is then redirected to the list page?
@arun-s-aot  @Josephalexantony-aot

File: forms-flow-web/src/components/Form/EditForm/FlowEdit.js [hunk 4]
```diff
@@ -108,6 +142,16 @@ const FlowEdit = forwardRef(({ isPublished = false, CategoryType, setWorkflowIsC
   const loadMoreBtnAction = () => {
     fetchHistories({ parentProcessKey: processData.parentProcessKey });
   };
+  const handleSaveFlowClick = () => {
+    if (shouldShowMigrationModal()) {
+      setShowMigrationModal(true);
```
Comment (shuhaib-aot) [inline forms-flow-web/src/components/Form/EditForm/FlowEdit.js:147]: could you please add some comment here, to understand what is happening

File: forms-flow-web/src/components/Form/EditForm/FlowEdit.js (hunk 3 above)
Comment (arun-s-aot) [reply forms-flow-web/src/components/Form/EditForm/FlowEdit.js:111]: its safer to route back to listing page as we dont have to handle any data descrepancy issues from edit page. As this is a one time operation per form, only user experience wll be impacted minimally

**Review by arun-s-aot — APPROVED** [review]

**Pull request merged**